src/spikeinterface/core/channelsaggregationrecording.py

```diff
@@ -16,16 +16,29 @@ class ChannelsAggregationRecording(BaseRecording):
     def __init__(self, recording_list, renamed_channel_ids=None):
         channel_map = {}
 
-        num_all_channels = sum([rec.get_num_channels() for rec in recording_list])
+        num_all_channels = sum(rec.get_num_channels() for rec in recording_list)
+
+        # Generate a default list of channel ids that are unique and consecutive numbers as strings.
+        default_channel_ids = [str(i) for i in range(num_all_channels)]
+
         if renamed_channel_ids is not None:
-            assert len(np.unique(renamed_channel_ids)) == num_all_channels, (
-                "'renamed_channel_ids' doesn't have the " "right size or has duplicates!"
-            )
+            assert (
+                len(np.unique(renamed_channel_ids)) == num_all_channels
+            ), "'renamed_channel_ids' doesn't have the right size or has duplicates!"
             channel_ids = list(renamed_channel_ids)
         else:
-            channel_ids = list(np.arange(num_all_channels))
+            # Collect channel IDs from all recordings
+            combined_ids = sum((list(rec.get_channel_ids()) for rec in recording_list), [])
```

Comment:
@h-mayorquin channel IDs are vector, so we can simply do:
```
# 1. check dtypes first
are_all_channels_same_type = np.unique([rec.channel_ids.dtype for rec in recording_list]).size == 1
if are_all_channels_same_type:
    combined_ids = np.concatenate([rec.channel_ids for rec in recording_list])
    # check that they are unique
    are_all_channels_unique = len(set(combined_ids)) == num_all_channels
    if are_all_channels_unique:
        channel_ids = combined_ids
    else:
        # If IDs are not unique or not of the same type, use default numeric IDs
        channel_ids = default_channel_ids
```

Reply:
Ok. I used numpy operators in the latest version.